sites/cnfpt/src/backend/base/tests/test_cache.py

```diff
@@ -0,0 +1,83 @@
+"""Test funmooc cache plugin"""
+
+from unittest import mock
+
+from django.core.cache.backends.dummy import DummyCache
+from django.test import TestCase, override_settings
+
+from django_redis.cache import RedisCache
+
+from base.cache import RedisCacheWithFallback
+
+
+def get_client(**kwargs):
+    return RedisCacheWithFallback(None, kwargs)
+
+
+def get_invalidation_fallback_needed(client):
+    return client._fallback_cache.get("FALLBACK_CACHE__INVALIDATION_REQUIRED")
+
+
+class RedisCacheWithFallbackTestCase(TestCase):
+    """
+    Test suite for the RedisCacheWithFallback
+
+    Credits:
+    - https://github.com/Kub-AT/django-cache-fallback
+    """
+
+    @override_settings(
+        CACHES={
+            "default": {
+                "BACKEND": "base.cache.RedisCacheWithFallback",
+                "LOCATION": "mymaster/redis-sentinel:26379,redis-sentinel:26379/0",
+                "OPTIONS": {
+                    "CLIENT_CLASS": "richie.apps.core.cache.SentinelClient",
+                },
+            },
+            "fallback": {
+                "BACKEND": "django.core.cache.backends.dummy.DummyCache",
+            },
+        }
+    )
+    def test_client(self):
+        """Test class instance of caches"""
+        client = get_client()
```

Comment:
I find this confusing (with self.client). Imo it's clearer to directly write this:
```suggestion
        cache = RedisCacheWithFallback(None, kwargs)
```